tests/Integration/goldens/spanner/src/V1/Client/DatabaseAdminClient.php

```diff
@@ -1042,28 +1042,21 @@ public function updateDatabaseDdl(UpdateDatabaseDdlRequest $request, array $call
     }
 
     /** Configure the gapic configuration to use a service emulator. */
-    private function getDefaultEmulatorConfig(): array
+    private function setDefaultEmulatorConfig(array $options): array
     {
         $emulatorHost = getenv('SPANNER_EMULATOR_HOST');
         if (empty($emulatorHost)) {
-            return [];
+            return $options;
         }
 
         if ($scheme = parse_url($emulatorHost, PHP_URL_SCHEME)) {
             $search = $scheme . '://';
             $emulatorHost = str_replace($search, '', $emulatorHost);
         }
 
-        return [
-            'apiEndpoint' => $emulatorHost,
-            'transportConfig' => [
-                'grpc' => [
-                    'stubOpts' => [
-                        'credentials' => ChannelCredentials::createInsecure(),
-                    ],
-                ],
-            ],
-            'credentials' => new InsecureCredentialsWrapper(),
-        ];
+        $options['apiEndpoint'] = $emulatorHost;
+        $options['transportConfig']['grpc']['stubOpts']['credentials'] = ChannelCredentials::createInsecure();
+        $options['credentials'] = new InsecureCredentialsWrapper();
```

Comment:
This looks great!! The only problem I see is that where before this was providing default values, it's now overwriting any user-supplied values. What if we use `??=` instead?

```suggestion
        $options['apiEndpoint'] ??= $emulatorHost;
        $options['transportConfig']['grpc']['stubOpts']['credentials'] ??= ChannelCredentials::createInsecure();
        $options['credentials'] ??= new InsecureCredentialsWrapper();
```

Reply:
Once the customer set the emulator environment variable, endpoint SHOULD be replaced by design, else it will by default hit the production product.

Reply:
I personally don't like that, I think they should just be defaults... I don't ever think an environment variable should REPLACE something in code - they should _always_ function as a default